### PR TITLE
feat(files): Rename files

### LINKED ISSUE #376: File: Add rename file option
### Add rename file option

### LINKED ISSUE #377: File: Remove filter from dialog and work with all type of files
### Remove filter from dialog and work with all type of files

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Add possibility to rename a file 
a. Moved Storage to ui_utils, to be possible to use inside File component, and with it needed to remove Eq derive from props 
b. File component now has some use_state and use_ref variables, because we need to rename it and update the component

### 2. New menu to options is now with context menu using mouse right click 
a. Old menu as just for test, this one is official, but download option not working yet

### 3. A key for file component was necessary to help Dioxus doesn't lost reference of items 
a. Without this key, ContextMenu lost reference and when upload a new file it comes from same name as the first one

### 4. Remove filter on file dialog to be possible upload any kind of file 
a. No big changes, just possible to upload other kind of files now

### Which issue(s) this PR fixes 🔨
- Resolve #376 
#377 

### Special notes for reviewers 🗒️
@phillsatellite Is possible to test rename with no problems, but it is missing an update yet from warp to not allow rename the same file in same directory with same name, the PR is already open, just need to be merged and this PR will be updated.

### CODE REVIEW

**Comment by phillsatellite** [comment]
Hi Lucas! Found a few things on here while testing,

1. When the User clicks rename, they should be able to start typing without having to click into the textbox first 

https://user-images.githubusercontent.com/93608357/204064974-e6b0f827-dd36-4d75-bf8e-34fa856006a0.mov

2. When the user clicks out of the textbox without typing anything or saving we should revert the file name back to what they had before. Similar to the behavior if the User navigates to a different page without typing or saving rename

https://user-images.githubusercontent.com/93608357/204065063-84ae35c5-5b49-4289-9b42-98c88f46d477.mov

3. If User rename's a file to the same name of a file they already have uploaded, they will not be able to delete either of them 

https://user-images.githubusercontent.com/93608357/204065110-0a41478e-dda8-4e40-ac55-4d3edd76ecbe.mov

**Review by dariusc93 — DISMISSED** [review]
I would suggest avoiding updating warp commit since i usually do testing when I made do a pr to update it here in uplink.

**Review by InfamousVague — APPROVED** [review]
This solution should have a // TODO: comment added to clean it up I think there is a better way to do this, but if this works as expected with QA it looks good to me for now.

**Comment by lgmarchi** [comment]
> I would suggest avoiding updating warp commit since i usually do testing when I made do a pr to update it here in uplink.

@dariusc93 Good to know Darius, I updated with last dev now.

> This solution should have a // TODO: comment added to clean it up I think there is a better way to do this, but if this works as expected with QA it looks good to me for now.

@InfamousVague I found other solution, clean, less code and working better with javascript

@phillsatellite Good to test again!

File: src/ui_kit/src/file/mod.rs [hunk 1]
```diff
@@ -19,19 +26,130 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
...
+            id: "{file_id}-file",
+                ContextMenu {
+                    parent: format!("{}-file", file_id.clone()),
+                    items: cx.render(
+                        rsx! {
+                                ContextItem {
+                                    icon: Shape::PencilSquare,
+                                    onpressed: move |_| {
+                                        // Use js script to show edit file name element
+                                        use_eval(&cx)(&show_edit_name_script);
```
Comment (dariusc93) [inline src/ui_kit/src/file/mod.rs:55]: Something to becareful about when doing it this way as replacing a section of the script while loading it in (as you are doing above) and evaluating it here could open possibly of an xss (though info wont be impacted but could still open holes if one is not careful with what its being replaced with, etc. Did leave a suggestion just adding a comment for it. Also using this function could panic if the js is invalid so something to becareful of too. 

```suggestion
                                        //TODO: Investigate in a way to replace use_eval in the future
                                        use_eval(&cx)(&show_edit_name_script);
```

File: src/ui_kit/src/file/mod.rs [hunk 2]
```diff
@@ -19,19 +26,130 @@ pub fn File<'a>(cx: Scope<'a, Props>) -> Element<'a> {
...
+                                *val.write_silent() = evt.value.to_string();
+                            },
+                            onkeyup: move |evt| {
+                                if evt.key_code == KeyCode::Enter {
+                                    let file_storage = cx.props.storage.clone();
+                                    let old_file_name = &*file_name_complete_ref.read();
+                                    let file_extension = cx.props.kind.clone();
+                                    let new_file_name = val.read();
+                                    // Use js script to hide edit file name element
+                                    use_eval(&cx)(&hide_edit_name_script);
```
Comment (dariusc93) [inline src/ui_kit/src/file/mod.rs:114]:
```suggestion
                                    //TODO: Investigate in a way to replace use_eval in the future
                                    use_eval(&cx)(&hide_edit_name_script);
```

**Review by dariusc93 — APPROVED** [review]
Looks good to me so far. Did leave a suggestion below though

File: src/ui_kit/src/file/mod.rs (hunk 1 above)
Comment (lgmarchi) [reply src/ui_kit/src/file/mod.rs:55]: Thanks Darius, I added TODO, something important to care. I used use_eval as recommendation from dioxus lab devs, maybe in master, should have better ways to run a js script.

I tested a invalid js, and the app didn't have panic, just not worked. 

https://user-images.githubusercontent.com/63157656/204306638-7b350a59-8dd7-4f2a-ae60-fe30a1856e34.mov

**Review by phillsatellite — APPROVED** [review]

**Pull request merged**